pkg/segment/reader/record/recordreader.go

```diff
@@ -37,9 +37,9 @@ import (
 )
 
 func ReadAllColsForRRCs(segKey string, vTable string, rrcs []*utils.RecordResultContainer,
-	qid uint64) (map[string][]utils.CValueEnclosure, error) {
+	qid uint64, deletedCols map[string]struct{}) (map[string][]utils.CValueEnclosure, error) {
```

Comment:
nit: I think renaming `deletedCols` to `ignoredCols` would be slightly better. This function doesn't need to know why it's skipping some columns; it just needs to know that it should skip some columns

Reply:
addressed in the latest commit.